### PR TITLE
Support Basic authentication challenges.

### PR DESCRIPTION
**Pull request by mattmoor**

### CODE REVIEW

File: client/v2/docker_http_.py [hunk 1]
```diff
@@ -161,8 +164,12 @@ def __init__(
     # Ping once to establish realm, and then get a good credential
     # for use with this transport.
     self._Ping()
-    if self._authenticated:
+    if self._authentication == _BEARER:
       self._Refresh()
+    elif self._authentication == _BASIC:
+      self._creds = self._basic_creds
+    else:
+      self._creds = docker_creds.Anonymous()
```
Comment (jmillikin-stripe) [inline client/v2/docker_http_.py]: Might be worth a `CheckState(self._authentication == _ANONYMOUS)` for sanity-checking and documentation.

**Comment by jmillikin-stripe** [comment]
LGTM

I'm sure you're aware, but the duplication of code between `v2` and `v2_2` is going to become a maintenance issue in the future. Consider ways that common code such as the auth process could be shared.

File: client/v2/docker_http_.py (hunk 1 above)
Comment (mattmoor) [inline client/v2/docker_http_.py]: Thanks, I'll add it to the source of truth.

**Comment by mattmoor** [comment]
Ack on the maintenance side.

**Pull request merged**